src/modules/custom.rs

```diff
@@ -77,26 +77,31 @@ pub fn module<'a>(name: &'a str, context: &'a Context) -> Option<Module<'a>> {
 
 /// Return the invoking shell, using `shell` and fallbacking in order to STARSHIP_SHELL and "sh"
 #[cfg(not(windows))]
-fn get_shell(shell: Option<&str>) -> std::borrow::Cow<str> {
-    if let Some(forced_shell) = shell {
-        forced_shell.into()
+fn get_shell<'a, 'b>(shell_args: &'b [&'a str]) -> (std::borrow::Cow<'a, str>, &'b [&'a str]) {
+    if !shell_args.is_empty() {
+        (shell_args[0].into(), &shell_args[1..])
     } else if let Ok(env_shell) = std::env::var("STARSHIP_SHELL") {
-        env_shell.into()
+        (env_shell.into(), shell_args)
     } else {
-        "sh".into()
+        ("sh".into(), shell_args)
     }
 }
 
 /// Attempt to run the given command in a shell by passing it as `stdin` to `get_shell()`
 #[cfg(not(windows))]
-fn shell_command(cmd: &str, shell: Option<&str>) -> Option<Output> {
-    let command = Command::new(get_shell(shell).as_ref())
+fn shell_command(cmd: &str, shell_args: &[&str]) -> Option<Output> {
+    let (shell, shell_args) = get_shell(shell_args);
+    let mut command = Command::new(shell.as_ref());
+
+    command
+        .args(shell_args)
         .stdin(Stdio::piped())
         .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .spawn();
+        .stderr(Stdio::piped());
+
+    handle_powershell(&mut command, &shell, shell_args);
```

Comment:
Well, I wasn't aware [we could run PowerShell on Linux](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-linux?view=powershell-7). That's pretty awesome, cool to see you got it cover here.

Reply:
Yep, and that's pretty new!